compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -6566,6 +6566,48 @@ boolean isNeverTypeOrStructureTypeWithARequiredNeverMember(BType type, Set<BType
         }
     }
 
+    public boolean containsClientObjectOrFunctionType(BType type, boolean isCheckForClient) {
+        Set<BType> visitedTypeSet = new HashSet<>();
+        visitedTypeSet.add(type);
+        return containsClientObjectOrFunctionType(type, visitedTypeSet, isCheckForClient);
+    }
+
+    boolean containsClientObjectOrFunctionType(BType type, Set<BType> visitedTypeSet, boolean isCheckForClient) {
+        if (type == null) {
+            return false;
+        }
+        type = Types.getReferredType(type);
```

Comment:
Before we call `containsClientObjectOrFunctionType`, we have get the referredType for `type`. So do we need to call `getReferredType` here as well?

Reply:
+1.
I will remove it

Reply:
Updated ebab2a74df0d4eddce0a69d7331556c045290bb9